app/api/items/navigation_data_mapper.go

```diff
@@ -58,20 +53,23 @@ func getRawNavigationData(dataStore *database.DataStore, rootID int64, user *dat
 	query := dataStore.Raw(`
 		SELECT items.id, items.type,
 			COALESCE(user_strings.title, default_strings.title) AS title,
-			attempts.id AS attempt_id,
-			attempts.score_computed AS score_computed, attempts.validated AS validated,
-			attempts.finished AS finished,
-			attempts.submissions AS submissions,
-			attempts.started_at AS started_at, attempts.validated_at AS validated_at,
+			IFNULL(best_scores.best_score, 0) AS best_score,
+			IFNULL(best_scores.validated, 0) AS validated,
 			items.child_order AS child_order,
 			items.content_view_propagation,
 			items.parent_item_id AS parent_item_id,
 			items.item_grandparent_id AS item_grandparent_id,
 			items.can_view_generated_value
 		FROM ? items`, itemThreeGenQ.SubQuery()).
 		JoinsUserAndDefaultItemStrings(user).
-		Joins("LEFT JOIN users_items ON users_items.item_id=items.id AND users_items.user_id=?", user.GroupID).
-		Joins("LEFT JOIN attempts ON attempts.id=users_items.active_attempt_id").
+		Joins(`
```

Comment:
As a max will always return a row, you can do a inner join, so not using this `ON 1` insanity ;-).
`JOIN LATERAL( ... ) AS best_scores`

Reply:
As the query contains 'GROUP BY' it still can return an empty set :(